2015/04/Upgrade_Insecure.md

```diff
@@ -0,0 +1,102 @@
+# Upgrade Insecure Requests Draft Feedback
+
+[Draft under discussion.](https://w3c.github.io/webappsec/specs/upgrade/) ([At this revision](https://github.com/w3c/webappsec/tree/f3ef3fb43394d57fa3ff9f93f22f3b057f7551be/specs/upgrade))
+
+## General Discussion
+
+This draft defines a new Content Security Policy header for a page to assert
+that subresources should be automatically upgraded to their secure version. We
+think this is a great step towards reducing barriers that website publishers
+face when transitioning from plaintext HTTP to secure connections.
+
+### ISSUE: Goal 1 Unclear
+
+Section 1.1, Goal 1:
+
+> Authors should be able to ensure that all content requested by a given page
+> loads successfully, and securely. Mixed content blocking should not break
+> pages as a result of migrating to a secure origin.
+
+This seems somewhat too ambitious for the spec. If third-party content on
+a page does not support HTTPS or stops supporting HTTPS, the page author cannot
+ensure that the content is loaded securely or at all. Inevitably, moving to
+a secure origin causes problems with mixed content blocking if the page has
+third party content that doesn't yet support HTTPS, a problem which the spec
+does not address.
+
+### ISSUE: Same-Origin vs Cross-Origin Behavior Unclear in Examples
+
+Talking to other TAG members about the spec, it became apparent that some of us
+thought the spec only applied upgrades to same-origin requests. I attribute
+most of my confusion to the examples in Section 1.2. Example #1 uses the
+example of `<img src="http://example.com/image.png">` being upgraded on
+`https://example.com` and Example #2 explicitly says that `<a
+href="http://not-example.com/">Home</a>` will *not* be upgraded on
+`https://example.com`. It would be better if Example #1 explicitly said that
+a third-party origin like `not-example.com` is upgradeable in that context, so
+that readers don't generalize Example #2 to all requests.
+
+### CLARIFICATION: Wording in Terminology
+
+The wording "depend on the upgrade-insecure-requests mechanism" in Section 2 is
+unclear. It seems to mean something like, "the same with and without
+upgrade-insecure-requests" from context, but I'm not sure.
+
+### COMMENT: +1 for Issue #184
+
+https://github.com/w3c/webappsec/issues/184 seems like a good thing for
+improving the smoothness of the HTTP to HTTPS transition.
+
+### ISSUE: Need Example for Upgrade Insecure Navigations Set
+
+It would be nice to have an example of a CSP directive with an upgrade insecure
+navigations set in the draft.
+
+### ISSUE: Upgrade All Navigations
+
+Is there a way to specify that all navigations should be upgraded? That seems
+useful if a webmaster doesn't want to change their header every time they add
+a link and are confident all their links will support HTTPS.
```

Comment:
This might fall out of w3c/webappsec#184 naturally, if we split the syntax between subresource and navigational upgrades in some way. It's not clear to me how valuable this is, given the lack of control over third-party sites, but I agree that it might be worth adding to cover some use cases.